server/src/main/java/org/opensearch/action/admin/cluster/shards/routing/weighted/delete/ClusterDeleteWeightedRoutingRequest.java

```diff
@@ -8,37 +8,142 @@
 
 package org.opensearch.action.admin.cluster.shards.routing.weighted.delete;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.OpenSearchGenerationException;
+import org.opensearch.OpenSearchParseException;
 import org.opensearch.action.ActionRequestValidationException;
 import org.opensearch.action.support.clustermanager.ClusterManagerNodeRequest;
+import org.opensearch.cluster.metadata.WeightedRoutingMetadata;
+import org.opensearch.common.bytes.BytesReference;
 import org.opensearch.common.io.stream.StreamInput;
 import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.DeprecationHandler;
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.common.xcontent.XContentType;
 
 import java.io.IOException;
+import java.util.Map;
 
 /**
  * Request to delete weights for weighted round-robin shard routing policy.
  *
  * @opensearch.internal
  */
 public class ClusterDeleteWeightedRoutingRequest extends ClusterManagerNodeRequest<ClusterDeleteWeightedRoutingRequest> {
-    public ClusterDeleteWeightedRoutingRequest() {}
+    private static final Logger logger = LogManager.getLogger(ClusterDeleteWeightedRoutingRequest.class);
+
+    private long version;
+    private String awarenessAttribute;
+
+    public void setVersion(long version) {
+        this.version = version;
+    }
+
+    ClusterDeleteWeightedRoutingRequest() {
+        this.version = WeightedRoutingMetadata.VERSION_UNSET_VALUE;
+    }
 
     public ClusterDeleteWeightedRoutingRequest(StreamInput in) throws IOException {
         super(in);
+        version = in.readLong();
```

Comment:
Will this fail for mixed version cluster tests?

Reply:
This is a breaking change. Delete request will fail if version is not provided.
Not sure if I understood the question correctly.

Reply:
Can we add a check to read these params only if OpenSearch version is 2.5 and above else there is a possibility that mixed version cluster tests will fail with serde issues. We might want to do the same in main as well.